### PR TITLE
Add ModalButtons, OverlayModal

### PR DESCRIPTION
**Pull request by winston0410**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,10 +34,13 @@
   "dependencies": {
     "@emotion/react": "11.7.1",
     "@mui/material": "5.3.1",
+    "@reduxjs/toolkit": "1.7.2",
```
Comment (vadim-kirpun) [inline package.json:37]: I think we can move it to peer since it will be installed anyway in MAA

File: package.json [hunk 2]
```diff
@@ -34,10 +34,13 @@
...
     "classnames": "^2.3.1",
     "lbh-frontend": "^3.6.1",
     "luxon": "2.3.0",
     "react-loader-spinner": "4.0.0",
+    "react-modal": "3.14.4",
+    "react-redux": "7.2.6",
```
Comment (vadim-kirpun) [inline package.json:43]: Same for react-redux

File: src/components/modals/ModalButtons/ModalButtons.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+);
+
+export const Normal = Template.bind({});
+Normal.args = {
+  withCancel: false,
+};
+
+export const WithCancel = Template.bind({});
+WithCancel.args = {
+  withCancel: true,
```
Comment (vadim-kirpun) [inline src/components/modals/ModalButtons/ModalButtons.stories.tsx:21]: `withCancel` is true by default so it is more like Normal case
Let's swap it with story above, so there are:
1) Normal - without arguments
2) WithoutCancel - withCancel=false

File: src/components/modals/ModalButtons/ModalButtons.test.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    expect(content).toBeInTheDocument();
+  });
+
+  it('renders cancel button withCancel', () => {
+    render(<ModalButtons withCancel />);
+
+    const content = screen.getByTestId('cancel-button');
+
+    expect(content).toBeInTheDocument();
```
Comment (vadim-kirpun) [inline src/components/modals/ModalButtons/ModalButtons.test.tsx:26]: Let's test `onConfirm` and `onCancel`
And `withCancel=false`

**Review by vadim-kirpun — CHANGES_REQUESTED** [review]

**Comment by vadim-kirpun** [comment]
Please fix the loader inside of a button so it doesn't increase the button's height
image

File: src/components/modals/OverlayModal/OverlayModal.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    <div>Overlay modal content</div>
+  </OverlayModal>
+);
+
+export const Open = Template.bind({});
+Open.args = {
+  isOpen: true,
+};
+
+export const Close = Template.bind({});
```
Comment (vadim-kirpun) [inline src/components/modals/OverlayModal/OverlayModal.stories.tsx:21]: Do we need this case? Story is empty

**Review by vadim-kirpun — CHANGES_REQUESTED** [review]

File: src/components/modals/OverlayModal/OverlayModal.stories.tsx (hunk 5 above)
Comment (winston0410) [reply src/components/modals/OverlayModal/OverlayModal.stories.tsx:21]: You mean removing the whole story???

Comment (vadim-kirpun) [reply src/components/modals/OverlayModal/OverlayModal.stories.tsx:21]: For `Close` case yes

**Review by vadim-kirpun — APPROVED** [review]

**Pull request merged**